docs/bots-guide.md

```diff
@@ -443,6 +443,103 @@ bot_handler.update_message(dict(
   development environment hosted on your computer, use
   `localhost:9991`
 
+### Writing tests for bots
+
+Bots in the [`python-zulip-api`](https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots)
+repository should have unittests. This section will explain the basics of writing a test suite for a bot.
+
+*Unittests for bots make heavy use of mocking. If you want to get comfortable with mocking, mocking strategies, etc.
+ you should check out our [mocking guide](testing-with-django.html#testing-with-mocks).*
+
+#### A simple example
+
+ Let's have a look at a simple test suite for the [`helloworld`](
+ https://github.com/zulip/python-zulip-api/tree/master/zulip_bots/zulip_bots/bots/helloworld) bot
+ (the actual test is written slightly more compact).
+
+     #!/usr/bin/env python
+
+    from __future__ import absolute_import
+    from __future__ import print_function
+
+    from six.moves import zip
+
+    from zulip_bots.test_lib import BotTestCase  # A base class for bot test suites that comes with various helpful testing methods.
+
+    class TestHelloWorldBot(BotTestCase):  # Our test suite for the bot.
+        bot_name = "helloworld"  # The bot's name (equivalent to the name of the bot module).
+
+        def test_bot(self): # A test case (must start with `test`)
+            # Messages we want to test and the expected bot responses.
+            message_response_pairs = {"" : "beep boop",
+                                      "foo" : "beep boop",
+                                      "Hi, my name is abc" : "beep boop"}
+            self.check_expected_responses(message_response_pairs)  # Test the bot with our message_response_pair dict.
+
+The `helloworld` bot replies with "beep boop" to every message @-mentioning it. Note that our helper method
+`check_expected_responses` adds the @-mention for us  - the only thing we need to do is to specify the
+rest of the message and the expected response. In this case, we want to assert that the bot always replies with "beep boop".
+To do so, we specify several test messages ("", "foo", "Hi, my name is abc") and assert that the response is always "beep boop".
```

Comment:
Your lines are generally too long; you should be wrapping at more like 80 characters.

Reply:
Also, I would spell it out a bit more: "and assert that the response is always correct, which for this simple bot, means always sending a reply with the content `beep boop`"